ethereum/core/src/main/java/tech/pegasys/teku/core/signatures/Signer.java

```diff
@@ -95,6 +96,18 @@ public Signer(final MessageSignerService signerService) {
     return signerService.signAggregateAndProof(signingRoot);
   }
 
+  public SafeFuture<BLSSignature> signVoluntaryExit(
+      final VoluntaryExit voluntaryExit, final ForkInfo forkInfo) {
+    final Bytes domain =
+        get_domain(
+            Constants.DOMAIN_VOLUNTARY_EXIT,
+            compute_epoch_at_slot(voluntaryExit.getEpoch()),
```

Comment:
You already have the epoch, so this is always generating invalid signatures. :)

Reply:
Oops. Good catch.

Reply:
Fixed.